java/compaction-core/src/main/java/sleeper/compaction/job/status/CompactionJobStatusesBuilder.java

```diff
@@ -26,28 +27,27 @@
 import java.util.stream.Stream;
 
 public class CompactionJobStatusesBuilder {
-    private final Map<String, CompactionJobStatusUpdateRecord> createdUpdateByJobId = new HashMap<>();
-    private final Map<String, List<CompactionJobStatusUpdateRecord>> runUpdatesByJobId = new HashMap<>();
+    private final Map<String, ProcessStatusUpdateRecord> createdUpdateByJobId = new HashMap<>();
+    private final ProcessStatusesBuilder processStatusesBuilder = new ProcessStatusesBuilder();
```

Comment:
Splitting out ProcessStatusesBuilder like this seems to make it a bit hard to see what it's doing. It'd be good if we can avoid that.